crates/cxx-qt-gen/src/parser/invokable.rs

```diff
@@ -28,3 +31,55 @@ pub struct ParsedQInvokable {
     /// Any specifiers that declared on the invokable
     pub specifiers: HashSet<ParsedQInvokableSpecifiers>,
 }
+
+impl ParsedQInvokable {
+    pub fn parse(method: &ImplItemMethod) -> Result<Self> {
```

Comment:
should we pass in the index ?

Reply:
As discussed lets move the stuff from cxxqtdata.rs into here. So then this can take a whole syn item, we'll do the same with signals etc.

Reply:
Lets do this here for now and consider a larger refactor later, as for example signals still need to ask the qobject btreemap if the T exists, so maybe with further thought something cleaner could be made.

I have wondered before about having our own syn tree that is fully parsed by syn instead.